apps/src/code-studio/progress.js

```diff
@@ -302,13 +304,16 @@ function initializeStoreWithProgress(
 
   store.dispatch(setIsAge13Required(scriptData.age_13_required));
 
-  // Progress from the server should be written down locally, unless we're a teacher
-  // viewing a student's work.
+  // Write progress locally unless we're a teacher viewing student work
   if (!isViewingStudentAnswer) {
     let lastProgress;
     store.subscribe(() => {
-      const nextProgress = store.getState().progress.levelProgress;
-      if (nextProgress !== lastProgress) {
+      const progressState = store.getState().progress;
+      const nextProgress = progressState.levelProgress;
+      const shouldUseDbProgress = progressState.shouldUseDbProgress;
+
+      // skip if user is logged in or progress hasn't changed
+      if (!(shouldUseDbProgress || nextProgress === lastProgress)) {
```

Comment:
Huh. You know, I agree. I had it that way and changed it because I figured there were fewer negations this way. Glad to hear you think the same way. I'll change it back :)

Reply:
:)